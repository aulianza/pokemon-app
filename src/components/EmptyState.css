.empty-state {
    text-align: -webkit-center;
}

.empty-state img {
    display: block;
    margin: 2rem;
}

.empty-state span {
    display: block;
    font-size: 1.2em;
}
