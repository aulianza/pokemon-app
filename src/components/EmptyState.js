import React from "react";
import NotFound from "../assets/images/empty.svg";
import styled from "@emotion/styled";

const EmptyState = () => {
    return (
        <StyledEmptyState>
            <img src={NotFound} alt="Not Found" width="200" height="200" />
            <span>No Pokemon Found.</span>
        </StyledEmptyState>
    );
};

export default EmptyState;

const StyledEmptyState = styled.div`
    text-align: -webkit-center;

    img {
        display: block;
        margin: 2rem;
    }

    span {
        display: block;
        font-size: 1.2em;
    }
`;
