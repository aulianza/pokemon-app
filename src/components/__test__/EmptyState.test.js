import EmptyState from '../EmptyState';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from "react-dom/test-utils";

let container = null;
beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

it("emptystate-component, snapshot", () => {
    act(() => {
        render(<EmptyState />, container)
    })
    expect(container).toMatchSnapshot();
})