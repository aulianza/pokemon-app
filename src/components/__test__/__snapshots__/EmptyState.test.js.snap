// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`emptystate-component, snapshot 1`] = `
<div>
  <div
    class="empty-state"
  >
    <img
      alt="Not Found"
      height="200"
      src="empty.svg"
      width="200"
    />
    <span>
      No Pokemon Found.
    </span>
  </div>
</div>
`;
